flask_mongoengine/pagination.py

```diff
@@ -1,37 +1,38 @@
 """Module responsible for custom pagination."""
 import math
 
-from flask import abort
 from mongoengine.queryset import QuerySet
 
+from flask_mongoengine.exceptions import InvalidPage
+
 __all__ = ("Pagination", "ListFieldPagination")
 
 
 class Pagination(object):
-    def __init__(self, iterable, page, per_page):
+    def __init__(self, iterable, page, per_page, max_depth=None):
```

Comment:
If you are working with class, please create docstrings with parameters description. For new and old.